packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -555,6 +555,22 @@ export async function activate(extensionContext: vscode.ExtensionContext) {
   // thus avoiding the potential errors surfaced when the libs call
   // process.cwd().
   ensureCurrentWorkingDirIsProjectPath(rootWorkspacePath);
+
+  // Check that the CLI is installed and that it is a supported version
+  // If there is no CLI or it is an unsupported version then the Core extension will not activate
+  const installed = await isCLIInstalled();
+  if (!installed) {
+    showCLINotInstalledMessage();
+  }
+  const cliVersionCheckResult = await new CheckCliVersion().validateCliVersion();
+  if (cliVersionCheckResult === 'cliNotSupported') {
+    showCLINotSupportedMessage();
+    throw new Error();
+  } else if (cliVersionCheckResult === 'cliNotInstalled') {
+    showCLINotInstalledMessage();
+    throw new Error();
+  }
```

Comment:
So keeping all the logic out of the activation method is a good thing (think separation of concerns).  Can we refactor this to be a stand alone utility function so all we see here is `await validateDependancies();`?  It will throw if all deps aren't correct and resolve happily otherwise.

Reply:
Also throwing empty errors is sadness (it's gets reported to telemetry).  We should throw a custom error or an error with a valid message